store/sqlstore/retention_policy_store.go

```diff
@@ -0,0 +1,521 @@
+	if len(teamIds) > 0 {
+		teamIdsMap := make(map[string]bool)
+		for _, teamId := range teamIds {
+			teamIdsMap[teamId] = false
+		}
+		teamsSelectQuery, teamSelectArgs, _ := s.getQueryBuilder().
+			Select("Id").
+			From("Teams").
+			Where(inStrings("Id", teamIds)).
+			ToSql()
+		var rows []*string
+		_, err := s.GetReplica().Select(&rows, teamsSelectQuery, teamSelectArgs...)
+		if err != nil {
+			return err
+		}
+		for _, teamId := range rows {
+			delete(teamIdsMap, *teamId)
+		}
+		for teamId := range teamIdsMap {
+			return store.NewErrNotFound("Team", teamId)
+		}
+	}
+	return nil
+}
+
+func (s *SqlRetentionPolicyStore) checkChannelsExist(channelIds []string) error {
+	if len(channelIds) > 0 {
+		channelIdsMap := make(map[string]bool)
+		for _, channelId := range channelIds {
+			channelIdsMap[channelId] = false
+		}
+		channelSelectQuery, channelSelectArgs, _ := s.getQueryBuilder().
+			Select("Id").
+			From("Channels").
+			Where(inStrings("Id", channelIds)).
+			ToSql()
+		var rows []*string
+		_, err := s.GetReplica().Select(&rows, channelSelectQuery, channelSelectArgs...)
+		if err != nil {
+			return err
+		}
+		for _, channelId := range rows {
+			delete(channelIdsMap, *channelId)
+		}
+		for channelId := range channelIdsMap {
+			return store.NewErrNotFound("Channel", channelId)
+		}
+	}
+	return nil
+}
+
+func (s *SqlRetentionPolicyStore) buildInsertRetentionPoliciesChannelsQuery(policyId string, channelIds []string) (query string, args []interface{}) {
+	if len(channelIds) > 0 {
+		builder := s.getQueryBuilder().
+			Insert("RetentionPoliciesChannels").
+			Columns("PolicyId", "ChannelId")
+		for _, channelId := range channelIds {
+			builder = builder.Values(policyId, channelId)
+		}
+		query, args, _ = builder.ToSql()
+	}
+	return
+}
+
+func (s *SqlRetentionPolicyStore) buildInsertRetentionPoliciesTeamsQuery(policyId string, teamIds []string) (query string, args []interface{}) {
+	if len(teamIds) > 0 {
+		builder := s.getQueryBuilder().
+			Insert("RetentionPoliciesTeams").
+			Columns("PolicyId", "TeamId")
+		for _, teamId := range teamIds {
+			builder = builder.Values(policyId, teamId)
+		}
+		query, args, _ = builder.ToSql()
+	}
+	return
+}
+
+func (s *SqlRetentionPolicyStore) Patch(patch *model.RetentionPolicyWithTeamAndChannelIds) (*model.RetentionPolicyWithTeamAndChannelCounts, error) {
+	// Strategy:
+	// 1. Update policy attributes
+	// 2. Delete existing channels in policy
+	// 3. Insert new channels into policy
+	// 4. Delete existing teams in policy
+	// 5. Insert new teams into policy
+	// 6. Read new policy
+	var (
+		rpUpdateQuery  string
+		rpUpdateArgs   []interface{}
+		rpcDeleteQuery string
+		rpcDeleteArgs  []interface{}
+		rpcInsertQuery string
+		rpcInsertArgs  []interface{}
+		rptInsertQuery string
+		rptInsertArgs  []interface{}
+		rptDeleteQuery string
+		rptDeleteArgs  []interface{}
+		rpSelectQuery  string
+		rpSelectProps  map[string]interface{}
+	)
```

Comment:
Also, these variable names are hard to distinguish between each other.  How about something like - 
```
channelDeleteQuery
teamDeleteQuery
```